doc/spire_server.md

```diff
@@ -96,6 +97,76 @@ The following configuration options are available to configure a plugin:
 
 Please see the [built-in plugins](#built-in-plugins) section below for information on plugins that are available out-of-the-box.
 
+## Federation configuration
```

Comment:
We should mention that both the `federation.bundle_endpoint` and `federation.federates_with` sections are both optional.